php/2020-SDK/Messaging/messaging-sendMMS-PHP/public/index.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+use BandwidthLib\Messaging\Models\BandwidthCallbackMessage;
+use BandwidthLib\Messaging\Models\BandwidthMessage;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Slim\Factory\AppFactory;
+
+require __DIR__ . '/../vendor/autoload.php';
+
+$BANDWIDTH_ACCOUNT_ID = getenv("BANDWIDTH_ACCOUNT_ID");
+$BANDWIDTH_API_USER = getenv("BANDWIDTH_API_USER");
+$BANDWIDTH_API_PASSWORD = getenv("BANDWIDTH_API_PASSWORD");
+$BANDWIDTH_MESSAGING_APPLICATION_ID = getenv("BANDWIDTH_MESSAGING_APPLICATION_ID");
+
+$config = new BandwidthLib\Configuration(
+    array(
+        "messagingBasicAuthUserName" => $BANDWIDTH_API_USER,
+        "messagingBasicAuthPassword" => $BANDWIDTH_API_PASSWORD
+    )
+);
+$client = new BandwidthLib\BandwidthClient($config);
+
+// Instantiate App
+$app = AppFactory::create();
+
+// Add error middleware
+$app->addErrorMiddleware(true, true, true);
+
+$messagingClient = $client->getMessaging()->getClient();
+
+$app->post('/outboundMessage', function (Request $request, Response $response) {
+  global $messagingClient, $BANDWIDTH_ACCOUNT_ID, $BANDWIDTH_MESSAGING_APPLICATION_ID;
+
+  $data = $request->getParsedBody();
+  $body = new BandwidthLib\Messaging\Models\MessageRequest();
+  $body->from = $data['from'];
+  $body->to = array($data['to']);
+  $body->applicationId = $BANDWIDTH_MESSAGING_APPLICATION_ID;
+  $body->text = $data['message'];
+  $body->media = $data['mediaUrl'];
+  try {
+      $msgResponse = $messagingClient->createMessage($BANDWIDTH_ACCOUNT_ID, $body);
+      $response->getBody()->write('{"Success":"Message sent successfully"}');
+      return $response->withStatus(201)
+        ->withHeader('Content-Type', 'application/json');
+  } catch (Exception $e) {
+      $response->getBody()->write('{"Error":"Message Failed"}');
+      return $response->withStatus(400);
+  }
+});
+
+$app->post('/messageCallback', function (Request $request, Response $response) {
+  global $messagingClient, $BANDWIDTH_ACCOUNT_ID;
+
+  $data = $request->getBody()->getContents();
+  $messagingCallbacks = \BandwidthLib\APIHelper::deserialize($data, BandwidthCallbackMessage::class, true );
+  $messageCallback = array_pop($messagingCallbacks);
+  $mediaName = $messageCallback->message->media;
+
+  // TODO: return each object from the callback
+  // var_dump($messageCallback)
+  $type = $messageCallback->type;
```

Comment:
DIdn't you have to do some conditionals based on the message type? Even if you're just handling one case, callbacks will still be sent to this URL for both inbound and outbound so I think at least the direction should be checked